test/DynamoCoreTests/Models/DynamoModelCommandsTest.cs

```diff
@@ -0,0 +1,294 @@
+
+            //This command is created with CreateAsDownstreamNode flas as true
+            var cmdOne = new DynamoModel.CreateAndConnectNodeCommand(newNodeGuid, existingNodeGuid,
+                "CoreNodeModels.CreateList", 0, 0, x, y, true, false);
+
+            //This command is created with CreateAsDownstreamNode flas as false
+            var cmdTwo = new DynamoModel.CreateAndConnectNodeCommand(newNodeGuid, existingNodeGuid,
+                "CoreNodeModels.CreateList", 0, 0, x, y, false, false);
+
+            //Act
+            //This will execute the method CreateAndConnectNodeImpl for both commands
+            CurrentDynamoModel.ExecuteCommand(cmdOne);
+            CurrentDynamoModel.ExecuteCommand(cmdTwo);
+
+            //Assert
+            Assert.IsNotNull(cmdOne);
+            Assert.IsNotNull(cmdTwo);
+        }
+
+        /// <summary>
+        /// This test method will execute the method GetNodeFromCommand from the DynamoModel class
+        /// The method has several flows then this test method will follow several flows
+        /// </summary>
+        [Test]
+        [Category("UnitTests")]
+        public void GetNodeFromCommandTest()
+        {
+            //Arrange
+            Guid newNodeGuid = Guid.NewGuid();
+
+            var command = new DynamoModel.CreateNodeCommand(newNodeGuid.ToString(), "NodeName1", 0, 0, true, true);
+            var command2 = new DynamoModel.CreateNodeCommand(newNodeGuid.ToString(), "CoreNodeModels.Watch", 0, 0, true, true);
+
+            //This will execute the proxy section code in the GetNodeFromCommand() method
+            var commandProxy = new DynamoModel.CreateProxyNodeCommand(newNodeGuid.ToString(), "CoreNodeModels.Watch", 0, 0, true, true,"ProxyCommand1",0,0);
+
+            //Assert
+            //Calling the ExecuteCommand internally executes the GetNodeFromCommand() method
+            Assert.Throws<Exception>( () => CurrentDynamoModel.ExecuteCommand(command));
+            CurrentDynamoModel.ExecuteCommand(command2);
+            Assert.Throws<Exception>(() => CurrentDynamoModel.ExecuteCommand(commandProxy));
+            
+        }
+
+        /// <summary>
+        /// This test method will execute the method GetNodeFromCommand from the DynamoModel class
+        /// It will follow the flow for when there is duplicate guid/name in the workspace
+        /// </summary> 
+        [Test]
+        [Category("UnitTests")]
+        public void GetNodeFromCommand_DuplicatedGuidsTest()
+        {
+            //Arrange
+            //This guid belongs to a CoreNodeModels.CreateList node inside the .dyn file
+            Guid existingNodeGuid = new Guid("81c94fd0-35a0-4680-8535-00aff41192d3");
+
+            string openPath = Path.Combine(TestDirectory, @"core\DetailedPreviewMargin_Test.dyn");
+            RunModel(openPath);
+
+            var command = new DynamoModel.CreateNodeCommand(existingNodeGuid.ToString(), "List.Create", 0, 0, true, true);  
```

Comment:
Is command `GetNodeFromCommand` part of `CreateNodeCommand`?

Reply:
@QilongTang 
The GetNodeFromCommand() method is called when the CreateNodeCommand is executed, then i added a explanatory comment about it in commit: https://github.com/DynamoDS/Dynamo/pull/10657/commits/e1fa29e30eef851af322f92301c437230be5ff30